src/content_script.js

```diff
@@ -179,120 +173,228 @@ function createBadgeFragment (socialAction) {
 
-    const htmlBadgeFragmentPromptContents = document.createElement("div");
-    htmlBadgeFragmentPromptContents.className = "fbc-badge-prompt-contents";
+  }
+
+}
 
-    for (let promptParagraphString of htmlEmailBadgeFragmentPromptParagraphStrings) {
-      const paragraph = document.createElement("p");
-      paragraph.appendChild(document.createTextNode(promptParagraphString));
-      htmlBadgeFragmentPromptContents.appendChild(paragraph);
-    }
 
-    const dontShowAgainCheckboxForm = document.createElement("div");
-    dontShowAgainCheckboxForm.className = "fbc-badge-prompt-dontShowAgain-checkbox";
-    const dontShowAgainCheckboxInput = document.createElement("input");
-    dontShowAgainCheckboxInput.type = "checkbox";
-    dontShowAgainCheckboxInput.id = "hideRelayEmailBadges";
-    dontShowAgainCheckboxInput.classList.add("fbc-badge-prompt-dontShowAgain-checkbox-input", "settings-checkbox");
-    const dontShowAgainCheckboxLabel = document.createElement("label");
-    dontShowAgainCheckboxLabel.htmlFor = "hideRelayEmailBadges";
-    const dontShowAgainCheckboxText = document.createTextNode( browser.i18n.getMessage("inPageUI-tooltip-prompt-checkbox") );
+function openInputPrompt(socialAction, fencePos, target, FBC_IFRAME_HEIGHT) {
 
-    dontShowAgainCheckboxLabel.appendChild(dontShowAgainCheckboxText);
-    dontShowAgainCheckboxForm.appendChild(dontShowAgainCheckboxInput);
-    dontShowAgainCheckboxForm.appendChild(dontShowAgainCheckboxLabel);
+  const iframeSrcVal = buildInpageIframe(socialAction, target, FBC_IFRAME_HEIGHT).src;
 
-    htmlBadgeFragmentPromptContents.appendChild(dontShowAgainCheckboxForm);
+  const hasFbcWrapper = document.querySelector(".fbc-wrapper");
+  if (!hasFbcWrapper) {
+    document.body.appendChild(injectIframeOntoPage(socialAction, target, FBC_IFRAME_HEIGHT));
+    positionIframe(fencePos);
+    ["resize", "scroll"].forEach(function (evt) {
+      if (document.querySelector(".fbc-wrapper")) {
+        window.addEventListener(evt, () => {
+          positionIframe(fencePos);
+        });
 
-    htmlBadgeFragmentPromptDiv.appendChild(htmlBadgeFragmentPromptContents);
+      }
+    });
 
-    const htmlBadgeFragmentPromptButtonDiv = document.createElement("div");
-    htmlBadgeFragmentPromptButtonDiv.className = "fbc-badge-prompt-buttons";
+    postMessageListeners(iframeSrcVal, target);
+    
+  } else {
+    hasFbcWrapper.remove();
+  }
 
-    for (let buttonString of htmlEmailBadgeFragmentPromptButtonStrings) {
-      const button = document.createElement("button");
-      button.className = "fbc-badge-prompt-" + buttonString;
-      button.appendChild(document.createTextNode( browser.i18n.getMessage(buttonString) ));
-      htmlBadgeFragmentPromptButtonDiv.appendChild(button);
+}
+
+function postMessageListeners(iframeSrcVal, target){
+
+  window.addEventListener("message", (e) => {
+    if (
+      e.data === "allowTriggered" 
+      && iframeSrcVal.includes(e.origin)
```

Comment:
Keeping the `postMessage` method since specified the origin is the iframe's src value. I did more reading up on it on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) and I saw this:

    If you do expect to receive messages from other sites, always verify the sender's identity using the origin and possibly source properties.